### DISCUSSION TITLE
405 Method Not Allowed on Vercel Deployment

### DISCUSSION THREAD

**Discussion by BluSimmon** [vercel/next.js#56282] [App Router]
This is my api function (only POST method):
```ts
import { NextResponse } from 'next/server'

export async function POST(req: Request) {
  const body = await req.json()

   ...

  return NextResponse.json({ success: true })
}
```
Everything work normally on my local environment
But when it is on vercel, i can receive 405 Method Not Allowed only

**Comment by Elindo586** [comment]
Did you fix this?

**Reply by BluSimmon** [reply]
Yes. I found that was the function in my route which has some issues.

**Reply by DutchPrince** [reply]
can you please elaborate? which function caused an issue? deploying to heroku works for me but vercel throws this same error
